soc/arm/st_stm32/common/pm_debug_jtag.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/init.h>
+
+#if !defined(CONFIG_DEBUG) && defined(CONFIG_PM)
```

Comment:
is `CONFIG_PM` a strict condition here?

Reply:
If `CONFIG_PM` is not a condition, it would mean that you can't use gdb unless CONFIG_DEBUG is set, which is not a condition today.
I already foresee that I will get tens of questions around this..
Using `CONFIG_PM`, I don't break anything as gdb can't be used today with CONFIG_PM unless CONFIG_DEBUG is set.